src/main/java/com/tabnine/inline/TabnineDocumentListener.java

```diff
@@ -26,7 +26,7 @@
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
-public class TabnineDocumentListener implements DocumentListener {
+public class TabnineDocumentListener implements BulkAwareDocumentListener {
```

Comment:
why changing to `BulkAware`? do we know when exactly does user's changes are applied in `bulk` mode? I think its not all the time, so we'll probably miss on many keystrokes.

Reply:
![image](https://user-images.githubusercontent.com/69103343/169760595-171ac475-ecf4-466c-a744-d7ad6911877b.png)
...

Reply:
Lets try this in alpha then, interesting